### ISSUE TITLE
`[RCS1264]` `roslynator_use_var = when_type_is_obvious` should recognize `StringBuilder`

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslynator#1435]
**Product and Version Used**: 

NuGet package v. 4.10.0

**Steps to Reproduce**:

in `.editorconfig`:
``` init
roslynator_use_var = when_type_is_obvious
```

Code:

``` csharp
internal class Demo
{
    public static string StringBuilderDemo(List<int> entries)
    {
        var sb = new StringBuilder("Entries: ")
            .Append(entries.Count)
            .Append('.');

        return sb.ToString();
    }
}
```

**Actual Behavior**:

On the `var` keyword: [`RCS1264`: Use 'var' or explicit type](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1264/)

**Expected Behavior**:

The StringBuilder is a well-known type, and this is not an uncommon pattern in my experience.

Much like `Parse(string)` methods, `StringBuilder`'s `Append` methods should be recognized as "obviously" returning `StringBuilder`.

**Comment by josefpihrt** [comment]
I know what you mean. It's true that in the particular code snippet the type is obvious.

The problem is that there are many code pattern similar to this one which could be called "type-obvious". And each one would require to write specific code to detect it.

Analyzers usually work is a much generic way which in this is a kind of syntax on the right side of a declaration which in this case is "invocation" (non-obvious) and not "object creation" (obvious).

If you don't want the code to look like
```
        StringBuilder sb = new StringBuilder("Entries: ")
            .Append(entries.Count)
            .Append('.');
```

possible option is to split declaration and Append calls:

```cs
        var sb = new StringBuilder("Entries: "); // no diagnostic here
        sb
            .Append(entries.Count)
            .Append('.');
```

The bottom line: it's doable but not worth the effort in my opinion.

**Comment by RenderMichael** [comment]
I understand what you mean, thank you for considering the issue. I would close this now but the labels added to this issue imply that roslynator would still accept this change. Is that correct? Should I leave the issue open?

**Comment by josefpihrt** [comment]
I would close it for now. If anybody is interested in this topic they can find this issue and contribute.
